plugins/destination/plugin.go

```diff
@@ -3,54 +3,127 @@ package destination
 import (
 	"context"
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
 	"github.com/rs/zerolog"
-	"golang.org/x/sync/errgroup"
+)
+
+type writerType int
+
+const (
+	unmanaged writerType = iota
+	managed
 )
 
 type NewClientFunc func(context.Context, zerolog.Logger, specs.Destination) (Client, error)
 
+type ManagedWriter interface {
+	PreWriteTableBatch(ctx context.Context, tables schema.Tables) error
+	WriteTableBatch(ctx context.Context, table *schema.Table, data [][]interface{}) error
+}
+
+type UnimplementedManagedWriter struct{}
+
+type UnmanagedWriter interface {
+	Write(ctx context.Context, tables schema.Tables, res <-chan *ClientResource) error
+	Metrics() Metrics
```

Comment:
Why is `Metrics()` part of `UnmanagedWriter`? Shouldn't it be part of the `Client` interface directly?

Reply:
Answered my own question by reading more of the code: looks like `Metrics()` doesn't get called for managed plugins but is implemented internally.